packages/block-editor/src/components/block-card/index.js

```diff
@@ -77,56 +105,85 @@ function BlockCard( {
 
 	const parentNavBlockClientId = useSelect(
 		( select ) => {
-			if ( ! allowParentNavigation ) {
+			if ( parentClientId || isChild || ! allowParentNavigation ) {
 				return;
 			}
-			const { getSelectedBlockClientId, getBlockParentsByBlockName } =
-				select( blockEditorStore );
-
-			const _selectedBlockClientId = getSelectedBlockClientId();
+			const { getBlockParentsByBlockName } = select( blockEditorStore );
 
 			return getBlockParentsByBlockName(
-				_selectedBlockClientId,
+				clientId,
 				'core/navigation',
 				true
 			)[ 0 ];
 		},
-		[ allowParentNavigation ]
+		[ clientId, allowParentNavigation, isChild, parentClientId ]
 	);
 
 	const { selectBlock } = useDispatch( blockEditorStore );
 
+	const TitleElement = parentClientId ? 'div' : 'h2';
+
 	return (
-		<div className={ clsx( 'block-editor-block-card', className ) }>
-			{ allowParentNavigation &&
-				parentNavBlockClientId && ( // This is only used by the Navigation block for now. It's not ideal having Navigation block specific code here.
-					<Button
-						onClick={ () => selectBlock( parentNavBlockClientId ) }
-						label={ __( 'Go to parent Navigation block' ) }
-						style={
-							// TODO: This style override is also used in ToolsPanelHeader.
-							// It should be supported out-of-the-box by Button.
-							{ minWidth: 24, padding: 0 }
-						}
-						icon={ isRTL() ? chevronRight : chevronLeft }
-						size="small"
-					/>
-				) }
-			<BlockIcon icon={ icon } showColors />
-			<VStack spacing={ 1 }>
-				<h2 className="block-editor-block-card__title">
-					<span className="block-editor-block-card__name">
-						{ !! name?.length ? name : title }
-					</span>
-					{ !! name?.length && <Badge>{ title }</Badge> }
-				</h2>
-				{ description && (
-					<Text className="block-editor-block-card__description">
-						{ description }
-					</Text>
-				) }
-				{ children }
-			</VStack>
+		<div
+			className={ clsx(
+				'block-editor-block-card',
+				{
+					'is-parent': parentClientId,
+					'is-child': isChild,
+				},
+				className
+			) }
+		>
+			{ parentNavBlockClientId && ( // This is only used by the Navigation block for now. It's not ideal having Navigation block specific code here.
+				<Button
+					onClick={ () => selectBlock( parentNavBlockClientId ) }
+					label={
+						parentNavBlockClientId
+							? __( 'Go to parent Navigation block' )
+							: // TODO - improve copy, not sure that we should use the term 'section'
+							  __( 'Go to parent section' )
```

Comment:
Since we're flattening the structure, could "go to parent" be enough? (Not a blocker for now, of course)

```suggestion
							  __( 'Go to parent' )
```

Reply:
Ended up removing this, it's from an earlier iteration!